src/isomorphic/classic/types/ReactPropTypes.js

```diff
@@ -342,6 +343,32 @@ function createShapeTypeChecker(shapeTypes) {
   return createChainableTypeChecker(validate);
 }
 
+function createFuncWithArityTypeChecker(expectedArity) {
+  var funcTypeCheck = createPrimitiveTypeChecker('function');
+
+  function validate(props, propName, componentName, location, propFullName) {
+    var funcTypeError = funcTypeCheck.apply(this, arguments);
```

Comment:
I'm unsure about this part. I like reusing the primitive `function` checker, but using `apply` might be too magical.